solidity/contracts/mERC20.sol

```diff
@@ -1,13 +1,15 @@
-// SPDX-License-Identifier: MIT
+// SPDX-License-Identifier: LGPL-3.0
 pragma solidity 0.8.29;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
 
-abstract contract AbstractERC20Upgradeable is ERC20Upgradeable, ERC20BurnableUpgradeable, Ownable2StepUpgradeable {
+abstract contract mERC20 is ERC20Upgradeable, ERC20BurnableUpgradeable, Ownable2StepUpgradeable {
     /// @dev The address of the minter.
     address public minter;
+    /// @dev The number of decimals of the token.
+    uint8 private _decimals;
```

Comment:
Why don't we make it public? We would avoid exposing the view function and do just `decimals = _decimals` in the initializer.

Reply:
Because we need to explicitly overwrite the function from the upstream contract.

Reply:
Ah, right.